### ISSUE TITLE
Plain JS files don't issue error on `await` inside non-async function

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#49483]
The following file should have an error in any JS file that isn't explicitly opted out of JS checking:

```js
function f() {
  await new Promise(undefined);
}
```

Expected: "await expressions are only allowed within async functions"
Actual: no error.

Note that other plain JS errors work fine; `function f() { break }` and `const x = 1; const x = 2` both error as expected.
